### PR TITLE
refactor: create command packages - part 1

### PR DESCRIPTION
**Pull request by zkochan**
ref #2154

- [x] every new registry mock should use its own storage in a system temp directory

### CODE REVIEW

**Comment by zkochan** [comment]
@pnpm/collaborators I am not sure how to call these packages.

One of them may implement several commands. So maybe prefix them with `plugin`?

* `@pnpm/plugin-publishing` - for `pnpm pack` and `pnpm publish`
* `@pnpm/plugin-listing` - for `pnpm list` and `pnpm why`
* `@pnpm/plugin-installation` - for `pnpm install/add/remove`
* etc

**Comment by bdchauvette** [comment]
I think the `@pnpm/plugin-` part makes sense. That's the pattern used by e.g. `@babel/plugin`, `@prettier/plugin`, `eslint-plugin` and `rollup-plugin`.

If the plugin system is intended to be used for things other than commands, than maybe keeping the `cli-` part from the current commits also makes sense:

- `@pnpm/plugin-cli-audit`
- `@pnpm/plugin-cli-publishing`
- etc.

Or maybe explicitly adding "command(s)":

- `@pnpm/plugin-audit-command`
- `@pnpm/plugin-publishing-commands`
- etc.

**Comment by ExE-Boss** [comment]
Or maybe:
- `@pnpm/plugin-command-audit`
- `@pnpm/plugin-commands-publishing`
- etc.

**Comment by zkochan** [comment]
Currently, it will only be about commands. But I guess prefixing all with `plugin-commands-` would work

File: packages/pnpm/src/cmd/index.ts [hunk 1]
```diff
@@ -1,16 +1,15 @@
+import * as audit from '@pnpm/cli-audit'
+import { list, why } from '@pnpm/cli-list-and-why'
+import * as outdated from '@pnpm/cli-outdated'
+import { pack, publish } from '@pnpm/cli-pack-and-publish'
```
Comment (ExE-Boss) [inline packages/pnpm/src/cmd/index.ts]: Maybe&nbsp;you&nbsp;could change&nbsp;this, so&nbsp;that you&nbsp;do:

```ts
import { audit } from '@pnpm/plugin-commands-audit'
import { list, why } from '@pnpm/plugin-commands-list-and-why'
import { outdated } from '@pnpm/plugin-commands-outdated'
import { pack, publish } from '@pnpm/plugin-commands-pack-and-publish'
```

or&nbsp;perhaps:
```ts
import audit from '@pnpm/plugin-commands-audit'
import { list, why } from '@pnpm/plugin-commands-list-and-why'
import outdated from '@pnpm/plugin-commands-outdated'
import { pack, publish } from '@pnpm/plugin-commands-pack-and-publish'
```

by&nbsp;introducing&nbsp;`index.ts` into&nbsp;`plugin‑commands‑audit` and&nbsp;`plugin‑commands‑outdated` that&nbsp;simply&nbsp;re‑exports the&nbsp;module&nbsp;namespace&nbsp;object of&nbsp;the&nbsp;actual CLI&nbsp;command implementation&nbsp;module as&nbsp;a&nbsp;named&nbsp;export.

Comment (zkochan) [reply packages/pnpm/src/cmd/index.ts]: I was planning to do the first one but using package names that don't contain `-and-`

```ts
import { audit } from '@pnpm/plugin-commands-audit'
import { list, why } from '@pnpm/plugin-commands-listing'
import { outdated } from '@pnpm/plugin-commands-outdated'
import { pack, publish } from '@pnpm/plugin-commands-publishing'
```

**Pull request merged**